src/cdk/v2/destinations/http/procWorkflow.yaml

```diff
@@ -36,24 +36,28 @@ steps:
     template: |
       const configAuthHeaders = $.getAuthHeaders(.destination.Config);
       const additionalConfigHeaders = $.getCustomMappings(.message, .destination.Config.headers);   
+      const metadataHeaders = $.metadataHeaders($.context.format);
       $.context.headers = {
         ...configAuthHeaders,
-        ...additionalConfigHeaders
+        ...additionalConfigHeaders,
+        ...metadataHeaders,
       }
 
   - name: prepareParams
     template: |
-      $.context.params = $.getCustomMappings(.message, .destination.Config.queryParams)
+      const params = $.getCustomMappings(.message, .destination.Config.queryParams)
+      $.context.params = $.encodeParamsObject(params)
 
   - name: deduceEndPoint
     template: |
-      $.context.endpoint = $.addPathParams(.message, .destination.Config.apiUrl);
+      $.context.endpoint = $.prepareEndpoint(.message, .destination.Config.apiUrl, .destination.Config.pathParams);
 
   - name: prepareBody
     template: |
       const payload = $.getCustomMappings(.message, .destination.Config.propertiesMapping);
-      $.context.payload = $.removeUndefinedAndNullValues($.excludeMappedFields(payload, .destination.Config.propertiesMapping))
-      $.context.format === "XML" &&  !$.isEmptyObject($.context.payload) ?  $.context.payload = {payload: $.getXMLPayload($.context.payload)};
+      $.context.payload = $.prepareBody(payload, .destination.Config.propertiesMapping, $.context.format)
+      $.context.payload = $.excludeMappedFields(payload, .destination.Config.propertiesMapping)
```

Comment:
this is duplicate code as it's already being handled in prepareBody

Reply:
Resolved